apm-sniffer/apm-sdk-plugin/resin-3.x-4.x-plugin/src/main/java/org/skywalking/apm/plugin/resin34x/ResinInterceptor.java

```diff
@@ -0,0 +1,65 @@
+package org.skywalking.apm.plugin.resin34x;
+
+import javax.servlet.http.HttpServletRequest;
+import org.skywalking.apm.agent.core.context.ContextCarrier;
+import org.skywalking.apm.agent.core.context.ContextManager;
+import org.skywalking.apm.agent.core.plugin.interceptor.EnhancedClassInstanceContext;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodInvokeContext;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.skywalking.apm.trace.Span;
+import org.skywalking.apm.trace.tag.Tags;
+import org.skywalking.apm.util.StringUtil;
+
+/**
+ * {@link ResinInterceptor} intercept method of{@link com.caucho.server.dispatch.ServletInvocation#service(javax.servlet.ServletRequest,
+ * javax.servlet.ServletResponse)} record the resin host, port ,url.
+ *
+ * @author baiyang
+ */
+public class ResinInterceptor implements InstanceMethodsAroundInterceptor {
+    /**
+     * Header name that the serialized context data stored in
+     * {@link HttpServletRequest#getHeader(String)}.
+     */
+    public static final String HEADER_NAME_OF_CONTEXT_DATA = "SWTraceContext";
+    /**
+     * Resin component.
+     */
+    public static final String RESIN_COMPONENT = "Resin";
+
+    @Override
+    public void beforeMethod(EnhancedClassInstanceContext context, InstanceMethodInvokeContext interceptorContext,
+        MethodInterceptResult result) {
+        Object[] args = interceptorContext.allArguments();
+        HttpServletRequest request = (HttpServletRequest)args[0];
+        Span span = ContextManager.createSpan(request.getRequestURI());
+        Tags.COMPONENT.set(span, RESIN_COMPONENT);
+        Tags.PEER_HOST.set(span, request.getServerName());
```

Comment:
Request#getServerName() cannot get the correct client host if Resin is behind a reverse proxy or load balancer.  You may need to use HttpServletRequest.getHeader("x-forwarded-proto") instead.